Fix: Correct quota per directory calculation

rolysr opened the pull request:
Previous changes related to adding quotas per directory as part of the volume size calculation did not correctly display the used space when a quota was set on the root directory. This issue was specifically observed on the Windows client. This commit resolves the issue.

rolysr committed:
fix: Correct quota per directory calculation

Previous changes related to adding quotas per directory as part of the
volume size calculation did not correctly display the used space when
a quota was set on the root directory. This issue was specifically
observed on the Windows client. This commit resolves the issue.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>

dmga44 commented on src/master/matoclserv.cc:
```diff
@@ -2054,16 +2054,37 @@ void matoclserv_sau_get_self_quota(matoclserventry *eptr, const uint8_t *data, u
+			results.emplace_back(
+				QuotaEntry{QuotaEntryKey{QuotaOwner{QuotaOwnerType::kInode, rootInode},
+											QuotaRigor::kUsed, QuotaResource::kSize},
+											rootInodeStatRec.size});
+		}
```
Please check if the size member of the `statsrecord` is the appropiate size of the displayed as used when mounting as a folder.

rolysr replied:
Done.

dmga44 reviewed (approved):
LGTM, please check my comment.

rolysr committed:
fix: Corrected stat behavior when using quotas

When the `usequotainvolumesize` option was enabled and quotas were set
for specific directories, executing the stat command on those
directories did not display the expected output, particularly for
total and free blocks. This commit resolves that issue.

Signed-off-by: Rolando Sánchez Ramos <rolysr@leil.io>
